### PR TITLE
upated bridge tapp to latest gradle version and removed Fabric depend…

### PR DESCRIPTION
**Pull request by siddutgikar**
…ency

### CODE REVIEW

File: android-wsbridge/.idea/compiler.xml [hunk 1]
```diff
@@ -1,22 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="CompilerConfiguration">
-    <resourceExtensions />
```
Comment (alainpimentel) [inline android-wsbridge/.idea/compiler.xml:4]: Remove the .idea folder and add to .gitignore

File: android-wsbridge/.idea/gradle.xml [hunk 2]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (alainpimentel) [inline android-wsbridge/.idea/gradle.xml]: File should be removed and ignored

File: android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java [hunk 3]
```diff
@@ -25,19 +24,13 @@ public static BridgeApplication getInstance() {
...
         super.onCreate();
-        if (BuildConfig.BUILD_TYPE != "debug") {
-            Fabric.with(this, new Crashlytics());
+        if(isInternalVersion()) {
+            FirebaseApp.initializeApp(getApplicationContext());
         }
         DJILogger.init();
         instance = this;
-        //registerActivityLifecycleCallbacks(this);
+        registerActivityLifecycleCallbacks(this);
```
Comment (alainpimentel) [inline android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java:33]: What is this needed for?

File: android-wsbridge/app/src/main/java/com/dji/wsbridge/BridgeActivity.java [hunk 4]
```diff
@@ -317,12 +321,12 @@ private void refreshRunners() {
                 deviceToWSRunner = new StreamRunner(wsInputStream, usbOutputStream, "Bridge to USB");
                 wsToDeviceRunner = new StreamRunner(usbInputStream, wsOutputStream, "USB to Bridge");
                 try {
-                    Crashlytics.log("Device to WS Runner alive "+ deviceToWSRunner.isAlive());
-                    Crashlytics.log("WS to Device Runner alive "+ wsToDeviceRunner.isAlive());
+                    logToFirebase("Device to WS Runner alive " + deviceToWSRunner.isAlive());
```
Comment (alainpimentel) [inline android-wsbridge/app/src/main/java/com/dji/wsbridge/BridgeActivity.java:323]: Is it ok to do Firebase call without initializing? For non internal builds

File: android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java [hunk 5]
```diff
@@ -51,6 +44,23 @@ public Bus getBus() {
         return this.eventBus;
     }
 
+    public static boolean isInternalVersion() {
```
Comment (alainpimentel) [inline android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java]: These would fit better in a separate utility class.

File: android-wsbridge/app/build.gradle [hunk 6]
```diff
@@ -40,28 +41,32 @@ android {
...
-    internalImplementation 'com.squareup.leakcanary:leakcanary-android:1.5.1'
-    releaseImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.1'
-    debugImplementation 'com.github.markzhai:blockcanary-android:1.5.0'
-    internalImplementation 'com.github.markzhai:blockcanary-android:1.5.0'
-    releaseImplementation 'com.github.markzhai:blockcanary-no-op:1.5.0'
+    implementation 'androidx.annotation:annotation:1.1.0'
+    implementation 'androidx.appcompat:appcompat:1.2.0'
+    implementation 'androidx.core:core:1.3.2'
+
+    implementation 'com.github.markzhai:blockcanary-android:1.5.0'
```
Comment (alainpimentel) [inline android-wsbridge/app/build.gradle:61]: Do we still need this? I see some blockcanary code was deleted in the BridgeApplication

**Review by alainpimentel — CHANGES_REQUESTED** [review]

File: android-wsbridge/app/src/main/java/com/dji/wsbridge/BridgeActivity.java (hunk 4 above)
Comment (siddutgikar) [reply android-wsbridge/app/src/main/java/com/dji/wsbridge/BridgeActivity.java:323]: The utility function is checking for initialization.

File: android-wsbridge/app/build.gradle (hunk 6 above)
Comment (siddutgikar) [reply android-wsbridge/app/build.gradle:61]: we still have a class using it.

File: android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java (hunk 5 above)
Comment (siddutgikar) [reply android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java]: Moved

File: android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java (hunk 3 above)
Comment (siddutgikar) [reply android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java:33]: I believe its only for the logs

File: android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java [hunk 7]
```diff
@@ -21,10 +23,22 @@ public static BridgeApplication getInstance() {
         return instance;
     }
 
+    public static void logToFirebase(String message) {
```
Comment (alainpimentel) [inline android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java]: I think these two should be in the Util class as well.

Comment (siddutgikar) [reply android-wsbridge/app/src/main/java/com/dji/wsbridge/lib/BridgeApplication.java]: Done

**Review by alainpimentel — APPROVED** [review]

**Pull request merged**